panic: runtime error: invalid memory address or nil pointer dereference in sshforward

```
time="2020-12-02T19:45:29Z" level=debug msg="> creating sc7ef2mq0r0z8sbyty3zci8xl [/bin/sh -c yarn install]"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc77423]

goroutine 4392 [running]:
github.com/moby/buildkit/session/sshforward.(*server).run.func2(0xc000920f68, 0x0)
	/src/session/sshforward/ssh.go:40 +0x1c3
golang.org/x/sync/errgroup.(*Group).Go.func1(0xc0041fa720, 0xc002d9d340)
	/src/vendor/golang.org/x/sync/errgroup/errgroup.go:57 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/src/vendor/golang.org/x/sync/errgroup/errgroup.go:54 +0x66
```

and I have no idea how to debug it or what logs should I provide.

I'm using `Docker version 20.10.0-rc1, build 5cc2396`, buildx from git master with
```sh
docker buildx stop
docker buildx rm
docker buildx create --name builder-build_test_scheduler-2 --driver docker-container --driver-opt image=moby/buildkit:v0.8.0-rc3 --buildkitd-flags --debug --use
docker buildx inspect --bootstrap
```
on linux x86_64, `docker buildx inspect` output is
```
Name:   builder-build_test_scheduler-2
Driver: docker-container

Nodes:
Name:      builder-build_test_scheduler-20
Endpoint:  unix:///var/run/docker.sock
Status:    stopped
Flags:     --debug
Platforms:
```


Trying to build private project, so no reproducer, unfortunately.

--- a/solver/llbsolver/mounts/mount.go
+++ b/solver/llbsolver/mounts/mount.go
@@ -178,6 +178,9 @@ func (mm *MountManager) getSSHMountable(ctx context.Context, m *pb.Mount, g sess
 	if err != nil {
 		return nil, err
 	}
+	if caller == nil {
+		return nil, nil
+	}
 	// because ssh socket remains active, to actually handle session disconnecting ssh error
 	// should restart the whole exec with new session
 	return &sshMount{mount: m, caller: caller, idmap: mm.cm.IdentityMapping()}, nil

